internal/monitors/kubernetes/cluster/metrics/node.go

```diff
@@ -40,9 +44,20 @@ func datapointsForNode(node *v1.Node, useNodeName bool) []*datapoint.Datapoint {
 		dims["machine_id"] = node.Status.NodeInfo.MachineID
 	}
 
-	return []*datapoint.Datapoint{
-		sfxclient.Gauge("kubernetes.node_ready", dims, nodeConditionValue(node, v1.NodeReady)),
+	datapoints := make([]*datapoint.Datapoint, 0)
+	for _, nodeConditionTypeValue := range nodeConditionTypesToReport {
+		v1NodeConditionTypeValue := v1.NodeConditionType(nodeConditionTypeValue)
+		nodeConditionMetric, ok := nodeConditionTypeToMetric[v1NodeConditionTypeValue]
```

Comment:
I would rather the metric name be constructed programatically (e.g. `fmt.Sprintf("kubernetes.node_%s", strcase.ToSnake(string(v1NodeConditionTypeValue)))`, we already use the `github.com/iancoleman/strcase` package in the agent so the dependency is free) instead so that we will automatically be able to support configuring conditions added in newer K8s versions or custom conditions.